src/MainWindow.vala

```diff
@@ -538,34 +548,37 @@ namespace Terminal {
                 current_terminal.grab_focus ();
             });
 
-            switch (Application.settings.get_string ("background")) {
-                case HIGH_CONTRAST_BG:
-                    color_button_white.active = true;
-                    break;
-                case SOLARIZED_LIGHT_BG:
-                    color_button_light.active = true;
-                    break;
-                case DARK_BG:
-                    color_button_dark.active = true;
-                    break;
-            }
-
-            color_button_dark.clicked.connect (() => {
+            color_button_dark.button_press_event.connect (() => {
                 Application.settings.set_boolean ("prefer-dark-style", true);
-                Application.settings.set_string ("background", DARK_BG);
-                Application.settings.set_string ("foreground", DARK_FG);
+                Terminal.Themes.set_active_name ("Default (Dark)");
+                if (color_preferences_dialog != null) {
+                    color_preferences_dialog.update_widgets_from_settings ();
+                }
+                return Gdk.EVENT_PROPAGATE;
             });
 
-            color_button_light.clicked.connect (() => {
+            color_button_light.button_press_event.connect (() => {
```

Comment:
Is there a reason for changing these from clicked to button press event?

Reply:
Actually yes :) I noticed that the `clicked` event for radio buttons is triggered once when the button is activated and (more surprising) once when the button is deactivated. This prevented the updating of the radio buttons from the color preferences dialog. This wasn't an issue until now since the order of the activating/deactivating of the different buttons was just right for this to work.

Reply:
That makes sense to me!